packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -315,6 +344,13 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
     [getRowId, columns],
   );
 
+  const handleGetRowHeight = React.useCallback(() => {
```

Comment:
@apedroferreira This is recalculating `hasImageColumns` for every call to `getRowHeight`. This could lead to O(nm) complexity, we can turn this O(1) if we do the following:

```tsx
  const getRowHeight = React.useMemo(() => {
    const hasImageColumns = columns.some(
      ({ customType }: ToolpadGridColDef) => customType === 'image',
    );
    return hasImageColumns ? () => 'auto' : undefined;
  }, [columns])
```

Reply:
Ah, I did not think of that, good point.
I'll add a fix in a separate PR.